### ISSUE TITLE
Template names are not fully considered

### ISSUE THREAD

**Issue by mwaldmueller** [Icinga/icingaweb2-module-graphite#145]
Templates names in Icinga 2 graphite feature with check_interval:

```
object GraphiteWriter "graphite" {
  ...
  host_name_template = "icinga2.$host.name$.host.$host.check_command$.$host.check_interval$"
  service_name_template = "icinga2.$host.name$.services.$service.name$.$service.check_command$.$service.check_interval$"
}
```

Template names in _/etc/icingaweb2/modules/graphite/config.ini_ and everything works fine:
```
  [icinga]
  graphite_writer_service_name_template = "icinga2.$host.name$.services.$service.name$.$service.check_command$.$service.check_interval$"
  graphite_writer_host_name_template = "icinga2.$host.name$.host.$host.check_command$.$host.check_interval$"
```

When the check_interval is changed, I get two trees on my filesystem:
```
# tree /opt/graphite/storage/whisper/icinga2/i2-master01_localdomain/services/load/ -L 3
/opt/graphite/storage/whisper/icinga2/i2-master01_localdomain/services/load/
└── load
    ├── 30
    │   ├── metadata
    │   └── perfdata
    └── 60
        ├── metadata
        └── perfdata
```

And graphs are not displayed any more:
<img width="549" alt="bildschirmfoto 2018-03-14 um 07 52 42" src="https://user-images.githubusercontent.com/3973350/37389752-c9f9b498-2765-11e8-9135-fcb168ab77cd.png">

Apache log:
```
10.37.129.2 - - [14/Mar/2018:07:48:42 +0100] "GET /icingaweb2/graphite/graph/service?service_name_template=icinga2.i2-master01_localdomain.services.load.load.60&host.name=i2-master01.localdomain&service.name=load&template=load&start=-3600&end=&width=632&height=220&cachebuster=99680919358935&legend=1 HTTP/1.1" 400 13928 "http://i2-master01.localdomain/icingaweb2/monitoring/host/services?host=i2-master01.localdomain&service=interfacetable" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_3) AppleWebKit/604.5.6 (KHTML, like Gecko) Version/11.0.3 Safari/604.5.6"
10.37.129.2 - - [14/Mar/2018:07:48:42 +0100] "GET /icingaweb2/graphite/graph/service?service_name_template=icinga2.i2-master01_localdomain.services.load.load.30&host.name=i2-master01.localdomain&service.name=load&template=load&start=-3600&end=&width=632&height=220&cachebuster=99680919401364&legend=1 HTTP/1.1" 400 13928 "http://i2-master01.localdomain/icingaweb2/monitoring/host/services?host=i2-master01.localdomain&service=interfacetable" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_13_3) AppleWebKit/604.5.6 (KHTML, like Gecko) Version/11.0.3 Safari/604.5.6"
```

## Environment
* Module version (System - About): graphite 0.9.0
* Icinga Web 2 version and modules (System - About): Icinga Web 2.5.1
* Icinga 2 version (`icinga2 --version`): Icinga 2.8.1
* Operating System and version: CentOS 7.4.1708
* Webserver, PHP versions: Apache (httpd) 2.4.6, PHP 7.1.8

**Comment by Al2Klimov** [comment]
Hello @mwaldmueller and thank you for reporting!

Unfortunately we support only host/service name and check command in this context.

As discussed w/ @dnsmichi (offline):
1. We could query the IDO, but this would require a mapping and would be not complete.
2. We could use (and cache) the Icinga 2 API, cmd. transport, but there's not neccessarily one configured – and there's sth. [WIP](https://github.com/Icinga/icingaweb2/issues/3035).

Anyway, this seems to be a really special case – what do you need this for?

Best,
AK

PS: Feel free to consult me offline.

**Comment by Al2Klimov** [comment]
As discussed offline:

Reasons for this setup:
1. The actual interval of the metric's values should be as close as possible to the interval expected by Graphite – to neither waste space nor lose data.
2. It's not as easy as 1-2-3 to change an existing whisper file's expected interval.

Consequence:
The Graphite mod can't resolve `$service.check_interval$`, so it lets it as-is and (later) treats it as ["service element" wildcard variable](https://github.com/Icinga/icingaweb2-module-graphite/blob/6caea7a23600fc534f3067af3c26df34bf6139d5/templates/disk.ini#L5) – you get your desired amount of graphs times the number of intervals of present data (e.g. 2 – 30 and 60).

**Comment by Al2Klimov** [comment]
@mwaldmueller I digged deeper in the code and discovered that such "wildcards" are just poorly supported if inside host|service_name_template. I thought whether I should fix it, but that would be 180° relative to the design of these two config options. These two shall contain only macros Icinga 2 provides concrete values for – they shall never be wildcards. The only design-conform solution would be actually substituting all of these macros w/ the actual concrete values from Icinga 2 (IDO/ API+cache).

## TL;DR

If the graph(s) worked before the interval change (only the old tree), they shall also work w/ only the new tree. Please try to drop the old tree if a new one is present – and tell your result.

**Comment by Al2Klimov** [comment]
@mwaldmueller Also this would be a great opportunity to sell development for those design-conform solution.

**Comment by mwaldmueller** [comment]
@Al2Klimov How much would you estimate to implement this?

**Comment by mwaldmueller** [comment]
Any updates?
